packages/table-core/src/features/RowSelection.ts

```diff
@@ -433,13 +441,25 @@ export const RowSelection: TableFeature = {
       )
     }
 
-    table.getIsSomePageRowsSelected = () => {
+    table.getIsSomePageRowsSelected = (ignoreCanSelect?: boolean) => {
       const paginationFlatRows = table.getPaginationRowModel().flatRows
-      return table.getIsAllPageRowsSelected()
+
+      const getIsSomeSelectedByCondition = () => {
+        return ignoreCanSelect
+          ? paginationFlatRows.some(
+              d => d.getIsSelected() || d.getIsSomeSelected()
+            ) &&
+              paginationFlatRows.some(
+                d => !d.getIsSelected() || !d.getIsSomeSelected()
+              )
+          : paginationFlatRows
+              .filter(row => row.getCanSelect())
+              .some(d => d.getIsSelected() || d.getIsSomeSelected())
+      }
```

Comment:
I think the readability is still lacking. 
How about separating this part into an independent function or clearly splitting it with if-else statements?